### ISSUE TITLE
How to use your repo?

### ISSUE THREAD

**Issue by shreejalt** [duyuanchao/adaptive-l2-regularization-pytorch#3]
Hi Thanks for the work @duyuanchao 
How can we use your repo with fast reid?
I saw there is a folder called adaptive L2 in `projects`. 
But can you tell me that if I want to train your model how can we du using fast reid?

**Comment by duyuanchao** [comment]
you can refer to https://github.com/JDAI-CV/fast-reid/blob/master/GETTING_STARTED.md

**Comment by shreejalt** [comment]
THanks @duyuanchao 
So in the paper there is these regional branches which they add. Can you point me in your repo that piece of code?

**Comment by duyuanchao** [comment]
I didnot implement the regional branches, this repo only contain the  adaptive-l2-loss

**Comment by shreejalt** [comment]
Oh okay.
Can you guide me in implementing it in your code? I think it will be a good add on.
If you can tell me what to change in your meta_wrapper, as I am unaware about fast-reid framework

**Comment by duyuanchao** [comment]
thanks a lot, that's very kind of you, 

you can refer to 

1. change the META_ARCHITECTURE, refer to 
```
https://github1s.com/JDAI-CV/fast-reid/blob/HEAD/fastreid/modeling/meta_arch/mgn.py
```
change these file
```
https://github1s.com/duyuanchao/adaptive-l2-regularization-pytorch/blob/master/projects/AdatpiveL2/meta_arch_wrapper.py
https://github1s.com/duyuanchao/adaptive-l2-regularization-pytorch/blob/master/projects/AdatpiveL2/backbone_wrapper.py
```

2.  refer to

```
https://github1s.com/JDAI-CV/fast-reid/blob/HEAD/configs/Base-SBS.yml
https://github1s.com/JDAI-CV/fast-reid/blob/HEAD/configs/Base-MGN.yml#L4
```
change this file
```
https://github1s.com/duyuanchao/adaptive-l2-regularization-pytorch/blob/master/projects/AdatpiveL2/configs/adaptive_l2_bot.yaml
```

**Comment by shreejalt** [comment]
Sure @duyuanchao 
I will do it and let you know.

Last doubt . So in meta_arch_wrapper you define `self.heads` and `self.backbone`, but you don't return `self.losses` in the forward function as it is done in the baseline meta_arch in fast-reid. 

Is it okay? Or while integrating it with fast-reid i need to add that?

**Comment by duyuanchao** [comment]
@shreejalt 

In fact I treat adaptive-l2-loss as a normlization method, so it doesnet the forward result, I calculate the adaptive-l2-loss in here
```
https://github.com/duyuanchao/adaptive-l2-regularization-pytorch/blob/HEAD/projects/AdatpiveL2/adaptive_l2_loss.py#L13
```
I donot have much confidence in this kind of implementation, what's your opinion
